tests/MongoDB.Driver.Core.Tests/Core/Operations/ChangeStreamCursorTests.cs

```diff
@@ -155,13 +211,154 @@ public void Dispose_can_be_called_more_than_once()
             mockBinding.Verify(s => s.Dispose(), Times.Once);
         }
 
+        [Theory]
+        [InlineData("{ a : 1 }", "{ b : 2 }", "{ c : 3 }", "{ d : 4 }", "{ a : 1 }")]
+        [InlineData(null, "{ b : 2 }", "{ c : 3 }", "{ d : 4 }", "{ b : 2 }")]
+        [InlineData(null, null, "{ c : 3 }", "{ d : 4 }", "{ c : 3 }")]
+        [InlineData(null, null, null, "{ d : 4 }", "{ d : 4 }")]
+        [InlineData(null, null, null, null, null)]
+        public void GetResumeToken_should_return_expected_result(
+            string postBatchResumeTokenJson,
+            string documentResumeTokenJson,
+            string startAfterJson,
+            string resumeAfterJson,
+            string expectedResult)
+        {
+            var mockCursor = CreateMockCursor();
+            var postBatchResumeToken = postBatchResumeTokenJson != null ? BsonDocument.Parse(postBatchResumeTokenJson) : null;
+            var documentResumeToken = documentResumeTokenJson != null ? BsonDocument.Parse(documentResumeTokenJson) : null;
+            var startAfter = startAfterJson != null ? BsonDocument.Parse(startAfterJson) : null;
+            var resumeAfter = resumeAfterJson != null ? BsonDocument.Parse(resumeAfterJson) : null;
+
+            var subject = CreateSubject(
+                cursor: mockCursor.Object,
+                postBatchResumeToken: postBatchResumeToken,
+                startAfter: startAfter,
+                resumeAfter: resumeAfter);
+            subject._documentResumeToken(documentResumeToken);
+            subject.GetResumeToken().Should().Be(expectedResult);
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void GetResumeToken_should_return_expected_results_when_batch_is_empty_or_fully_iterated(
```

Comment:
Prose tests:
11. For a ChangeStream under these conditions:
            Running against a server >=4.0.7.
            The batch is empty or has been iterated to the last document.

    Expected result:
            getResumeToken must return the postBatchResumeToken from the current command response.

12. For a ChangeStream under these conditions:
            Running against a server <4.0.7.
            The batch is empty or has been iterated to the last document.

    Expected result:
            getResumeToken must return the _id of the last document returned if one exists.
            getResumeToken must return startAfter from the initial aggregate if the option was specified.
            getResumeToken must return resumeAfter from the initial aggregate if the option was specified.
            If neither the startAfter nor resumeAfter options were specified, the getResumeToken result must be empty.